src/org/etools/j1939_84/controllers/part03/Part03Step16Controller.java

```diff
@@ -54,12 +58,41 @@
     @Override
     protected void run() throws Throwable {
         // 6.3.16.1.a. Turn the engine off.
+        incrementProgress("Part 3, Step 16 Turn Engine Off and keep the ignition key in the off position");
+        ensureKeyOffEngineOff();
+
         // 6.3.16.1.b. Confirm Fault A is still implanted according to the manufacturer’s instruction.
+        incrementProgress("Waiting for implant of Fault A according to the engine manufacturer's instruction");
+        waitForFault("Part 6.3.16.1.b");
```

Comment:
This shouldn't wait for a fault.  It should just have the your press "OK or Cancel"

Reply:
Per your reqeust:
 // 6.3.16.1.b. Confirm Fault A is still implanted according to the manufacturer’s instruction.
        incrementProgress("Please confirm Fault A is still implanted according to the manufacturer's instruction");
        waitForFault("Part 6.3.16.1.b");

Reply:
This Step (unlike the last) isn't waiting for a fault to be implanted.  It's waiting per the manufacturer's instructions.

Reply:
This needs to display an OK/Cancel dialog indicating we are waiting according to the manufacturer's instructions.

Reply:
That is c.
 // 6.3.16.1.c. Wait manufacturer’s recommended interval with the key in the off position.
        incrementProgress("Waiting for manufacturer's recommended interval after the Fault A implant");
        waitForManufacturerInterval();

Reply:
6.3.16.1.b is about the user confirming the fault is still implanted, not waiting.

The `waitForFault()` method will display:
> Implant Fault A according to engine manufacturer’s instruction
> Press OK when ready to continue testing

That's not what we asking the user to do at this point.  We are asking them to confirm the fault still exists.